sql/src/main/java/io/crate/operation/operator/CmpOperator.java

```diff
@@ -48,10 +52,19 @@ public Symbol normalizeSymbol(Function symbol) {
         if (left.symbolType().isValueSymbol() && right.symbolType().isValueSymbol()) {
             // must be true due to the function registration (argument DataType signature)
 
-            if (compare(((Literal) left).compareTo((Literal)right))) {
-                return Literal.newLiteral(true);
+            boolean equal = false;
+            try {
+                equal = compare(((Literal) left).compareTo((Literal) right));
+            } catch (ClassCastException e){
```

Comment:
In which case can the ClassCastException happen here? CmpOperator should only be used if both types match?